File: ex1/answers/problem2.py
#!/usr/bin/env python

def is_prime(n):
    if n < 2:
        return False
    prime = True
    for i in range(2,n):
        if n % i == 0:
            prime = False
            break
    return prime

def main():
    n_primes = 0
    i = 0
    while n_primes < 10000:
        i += 1
        if is_prime(i):
            print(i)
            n_primes += 1

if __name__ == "__main__":
    main()

File: ex1/answers/problem2.c
#include <stdio.h>

// Return 1 if n is prime, and return 0 otherwise
int
is_prime (int n)
{
  int d;
  int prime = 1;
  if (n < 2) {
    return 0;
  }
  for (d=2;d<n;d++) {
    if (n%d == 0) {
      prime = 0;
      break;
    }
  }
  return prime;
}

// Find the first 1000 primes.
int
main (void)
{
  int n_primes = 0;
  int i = 0;
  while (n_primes < 10000) {
    i++;
    if (is_prime(i)){
      printf("%d\n", i);
      n_primes++;
    } 
  }
}
